COPYRIGHT.txt

```diff
@@ -55,48 +55,35 @@ Comment: Godot Engine logo
 Copyright: 2017, Andrea Calabró
 License: CC-BY-3.0
 
-Files: ./platform/android/java/aidl/com/android/vending/billing/IInAppBillingService.aidl
- ./platform/android/java/res/layout/status_bar_ongoing_event_progress_bar.xml
- ./platform/android/java/src/com/google/android/vending/expansion/downloader/*
- ./platform/android/java/src/com/google/android/vending/licensing/*
- ./platform/android/java/src/org/godotengine/godot/input/InputManagerCompat.java
- ./platform/android/java/src/org/godotengine/godot/input/InputManagerV16.java
+Files: ./platform/android/java/lib/aidl/com/android/*
+ ./platform/android/java/lib/res/layout/status_bar_ongoing_event_progress_bar.xml
+ ./platform/android/java/lib/src/com/google/android/*
+ ./platform/android/java/lib/src/org/godotengine/godot/input/InputManagerCompat.java
+ ./platform/android/java/lib/src/org/godotengine/godot/input/InputManagerV16.java
 Comment: The Android Open Source Project
-Copyright: 2008-2013, The Android Open Source Project
+Copyright: 2002, Google, Inc.
+Copyright: 2008-2016, The Android Open Source Project
 License: Apache-2.0
 
-Files: ./platform/android/java/src/com/android/vending/licensing/util/Base64.java
- ./platform/android/java/src/com/android/vending/licensing/util/Base64DecoderException.java
-Comment: The Android Open Source Project
-Copyright: 2002, Google Inc.
-License: Apache-2.0
-
-Files: ./scene/animation/tween_interpolaters.cpp
-Comment: Penner Easing
-Copyright: 2001, Robert Penner
-License: BSD-3-clause
-
-Files: ./servers/physics/gjk_epa.cpp
- ./servers/physics/joints/generic_6dof_joint_sw.cpp
- ./servers/physics/joints/generic_6dof_joint_sw.h
- ./servers/physics/joints/hinge_joint_sw.cpp
- ./servers/physics/joints/hinge_joint_sw.h
- ./servers/physics/joints/jacobian_entry_sw.h
- ./servers/physics/joints/pin_joint_sw.cpp
- ./servers/physics/joints/pin_joint_sw.h
- ./servers/physics/joints/slider_joint_sw.cpp
- ./servers/physics/joints/slider_joint_sw.h
+Files: ./servers/physics_3d/gjk_epa.cpp
+ ./servers/physics_3d/joints/generic_6dof_joint_3d_sw.cpp
+ ./servers/physics_3d/joints/generic_6dof_joint_3d_sw.h
+ ./servers/physics_3d/joints/hinge_joint_3d_sw.cpp
+ ./servers/physics_3d/joints/hinge_joint_3d_sw.h
+ ./servers/physics_3d/joints/jacobian_entry_3d_sw.h
+ ./servers/physics_3d/joints/pin_joint_3d_sw.cpp
+ ./servers/physics_3d/joints/pin_joint_3d_sw.h
+ ./servers/physics_3d/joints/slider_joint_3d_sw.cpp
+ ./servers/physics_3d/joints/slider_joint_3d_sw.h
 Comment: Bullet Continuous Collision Detection and Physics Library
 Copyright: 2003-2008, Erwin Coumans
- 2007-2020, Juan Linietsky, Ariel Manzur.
```

Comment:
It's not super important but those files have Godot made changes and use the same license header as the rest of the engine, and this line is part of Godot's copyright statement.

Reply:
These files were added in 2014 i.e. after it was made open source; so the changes should be covered under the next line:
```2014-2020, Godot Engine contributors.```

Reply:
We don't go in this kind of details on a per-file basis. New files added in 2020 also get the same copyright header as in `LICENSE.md`.

Moreover those files date back to before the opensourcing:
```
commit 0b806ee0fc9097fa7bda7ac0109191c9c5e0a1ac
Author: Juan Linietsky <reduzio@gmail.com>
Date:   Sun Feb 9 22:10:30 2014 -0300

    GODOT IS OPEN SOURCE
```

Reply:
Sorry, I edit my previous comment after your comment; so I'll repeat my points here:
Except for `gjk_epa.cpp` the files were added with 8cab401.
Also, I'm sure Erwin would challenge any claim to copyright earlier than 2008.

Reply:
Looking at the copyright statement on the individual files, it states:
> This file is part of: GODOT ENGINE

So it makes sense to have an overall copyright statement for the Godot engine:
> Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.
> Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).

Whereas here we are listing specific files, so it makes sense to be more precise.

Looking at the `gjk_eps.cpp` file, I suppose it's the last lines that would fall under Juan and Ariel's copyright, because they were added before 2014:
https://github.com/godotengine/godot/blob/0b806ee0fc9097fa7bda7ac0109191c9c5e0a1ac/servers/physics/gjk_epa.cpp#L884-L900

Reply:
I'm not sure what's the purpose of this nitpicking, but those files have been heavily modified to build using Godot's math and physics types, which is why they are still in `servers/physics` and not in `thirdparty`.
Since they include Godot code, they fall under the Godot copyright for the **for the code that was added**. The copyright header does not apply the entirety of the code, it simply notifies that the contents of this file are partly under their original copyright, and partly under Godot copyright.

Now, if you want to compare with upstream and get rid of all Godot-made modifications, I'd be happy to see those files as unmodified thirdparty code put in `thirdparty`, and then you can indeed remove the Godot copyright header.

If you want to challenge that Godot has any copyright on part of those files, then you should remove the whole Godot copyright header, not just Juan's line.

Reply:
I'm all for removing the copyright header if the modifications to these files are not substantial.

In this case, those modifications need to be documented (ideally with a `.patch` file in the repository, as we do for all other modifications to thirdparty code).

Reply:
All I can say is that, I once read the copyright statements directly in Godot out of curiosity, and got surprised by this one... So perhaps worth mentioning that these are `Modifications Copyright`, or adding some kind of brief summary of changes, else it's just misleading in either case. But please ignore if what I say is silly in terms of copyright law. 🙂